packages/devtools/lib/src/framework/framework.dart

```diff
@@ -30,23 +39,108 @@ class Framework {
 
     connectDialog = new ConnectDialog(this);
 
+    importMessage = new ImportMessage(this);
+
     analyticsDialog = AnalyticsOptInDialog(this);
   }
 
   final List<Screen> screens = <Screen>[];
 
+  final Map<Screen, CoreElement> _screenContents = {};
+
   final Completer<void> screensReady = Completer();
 
   Screen current;
 
+  Screen previous;
+
   StatusLine globalStatus;
   StatusLine pageStatus;
   StatusLine auxiliaryStatus;
   ActionsContainer globalActions;
   ConnectDialog connectDialog;
+  ImportMessage importMessage;
   AnalyticsOptInDialog analyticsDialog;
 
-  final Map<Screen, CoreElement> _screenContents = {};
+  void _initDragDrop() {
+    window.addEventListener('dragover', (e) => _onDragOver(e), false);
+    window.addEventListener('drop', (e) => _onDrop(e), false);
+  }
+
+  void _onDragOver(MouseEvent event) {
+    // This is necessary to allow us to drop.
+    event.preventDefault();
+    event.dataTransfer.dropEffect = 'move';
+  }
+
+  void _onDrop(MouseEvent event) async {
+    // Stop the browser from redirecting.
+    event.preventDefault();
+
+    final List<File> files = event.dataTransfer.files;
+    if (files.length > 1) {
+      toast('You cannot import more than one file.');
+      return;
+    }
+
+    final droppedFile = files.first;
+    if (droppedFile.type != 'application/json') {
+      toast('${droppedFile.type} is not a supported file type.\n Please import '
+          'a .json file.');
+      return;
+    }
+
+    final FileReader reader = FileReader();
+    reader.onLoad.listen((_) {
+      final Map<String, dynamic> import = jsonDecode(reader.result);
+      final source = import['source'];
+
+      // TODO(jacobr): add the inspector handling case here once the inspector
+      // can be exported.
+      switch (source) {
+        case timelineScreenId:
+          TimelineScreen timelineScreen = screens.firstWhere(
+            (screen) => screen.id == timelineScreenId,
+            orElse: () => null,
+          );
+          if (timelineScreen == null) {
+            addScreen(timelineScreen = TimelineScreen(disabled: false));
+          }
+          connectDialog.hide();
+          importMessage.hide();
+          _importMode = true;
+          navigateTo(timelineScreenId);
+          timelineScreen.loadFromImport(import);
+          break;
+        default:
+          toast(
+            'The imported file is from an unrecognized source page: $source. '
```

Comment:
How about show a different message if `source == null` as that case means someone dragged in
a json file that probably wasn't from devtools.
also change source from`source` to `dartDevtoolsScreen` or something else unique so that there is no doubt if that key is in the json it was because they opened a devtools file.
The reason I think we should care about this is that at some point someone will try to open a devtools file from a newer devtools version in an older devtools and then wonder why it doesn't work. Similarly people will try to open chrome trace files or other not quite compatible json files in devtools.

Reply:
done.